Dockerfile

```diff
@@ -0,0 +1,54 @@
+FROM ubuntu:trusty
+MAINTAINER Di Wu <diwu@yelp.com>
+
+COPY . /opt/redash/current/
+
+WORKDIR /opt/redash/current
+
+ENV FILES_BASE_URL /opt/redash/current/setup/files/
+
+# Base packages
+RUN apt-get dist-upgrade
+RUN apt-get update
+RUN apt-get install -y python-pip python-dev curl build-essential pwgen libffi-dev sudo git-core
```

Comment:
Contrary to what you would expect, it's a best practice to merge multiple RUN commands into one multi command. So in this case it will become:

```
RUN apt-get update \
         apt-get install -y ...
```

This to avoid bloat of multiple layers that won't be really used.

So it's hard to point what commands you should group together, and it's left to the Docker file creator to decide, but I would take all the apt-get commands from this file and group them into one RUN command.